File: verdant-frontend/src/lib/types/component-registry.ts
export type ComponentRegistryItem = {

    id: string;
    equipment_id: string;
    component_type: string;
    component_name: string;
    manufacturer?: string;
    model_number?: string;
    serial_number?: string;
    status: string;
    installation_date?: string;
    replacement_date?: string;
    warranty_expiration?: string;
    useful_life_months?: string;
    failure_reason?: string;
    replacement_reason?: string;
    notes?: string;
    is_critical: boolean;
    maintenance_log_id?: string;
    created_at: string;
    equipment_asset_tag?: string;
    equipment_model?: string;
    health_score: number;
    warranty_status: string;
    mtbf_days?: number;
    failure_events: number;
};

export type ComponentRegistrySnapshot = {

    total_components: number;
    critical_components: number;
    failed_components: number;
    warranty_expiring: number;
    average_health_score: number;
    items: ComponentRegistryItem[];
};

File: verdant_erp_backend/app/schemas/hvac/component_registry.py
from datetime import date
from datetime import datetime
from pydantic import BaseModel
from uuid import UUID


class ComponentRegistryItemDTO(BaseModel):

    id: UUID
    equipment_id: UUID
    component_type: str
    component_name: str
    manufacturer: str | None = None
    model_number: str | None = None
    serial_number: str | None = None
    status: str
    installation_date: date | None = None
    replacement_date: date | None = None
    warranty_expiration: date | None = None
    useful_life_months: str | None = None
    failure_reason: str | None = None
    replacement_reason: str | None = None
    notes: str | None = None
    is_critical: bool
    maintenance_log_id: UUID | None = None
    created_at: datetime
    equipment_asset_tag: str | None = None
    equipment_model: str | None = None
    health_score: int
    warranty_status: str
    mtbf_days: int | None = None
    failure_events: int


class ComponentRegistrySnapshotDTO(BaseModel):

    total_components: int
    critical_components: int
    failed_components: int
    warranty_expiring: int
    average_health_score: int
    items: list[ComponentRegistryItemDTO]